yesod-core/src/Yesod/Routes/Class.hs

```diff
@@ -16,10 +20,49 @@ class Eq (Route a) => RenderRoute a where
     renderRoute :: Route a
                 -> ([Text], [(Text, Text)]) -- ^ The path of the URL split on forward slashes, and a list of query parameters with their associated value.
 
+-- | This class acts as a delegation class for 'RenderRoute' on nested
+-- route fragments.
+--
+-- @since 1.6.28.0
+class Eq a => RenderRouteNested a where
+    -- | Render the fragment of the route. To form a complete route, you'll
+    -- need to `mappend` this with the result from the parent `renderRoute`
+    -- or `renderRouteNested` call.
+    --
+    -- @since 1.6.28.0
+    renderRouteNested :: a -> ([Text], [(Text, Text)])
+
 class RenderRoute a => ParseRoute a where
     parseRoute :: ([Text], [(Text, Text)]) -- ^ The path of the URL split on forward slashes, and a list of query parameters with their associated value.
                -> Maybe (Route a)
 
+-- | Like 'RenderRouteNested', this acts as a delegation class for nested
+-- route fragments.
+--
+-- @since 1.6.28.0
+class RenderRouteNested a => ParseRouteNested a where
+    -- |
+    --
+    -- @since 1.6.28.0
+    parseRouteNested
+        :: ([Text], [(Text, Text)])
+        -- ^ The path of the URL split on forward slashes, and a list of query parameters with their associated value.
+        --
+        -- Unlike for 'parseRoute', this will not include URL fragments as
+        -- part of parent routes. It is expected that parents will take the
+        -- resulting @a@ and construct the parent from that.
+        -> Maybe a
```

Comment:
Meanwhile, this... hmm. Seems like `renderRouteNested` and `parseRouteNested` should have inverse behavior. So if `renderRouteNested` knows about parent stuff, then `parseRouteNested` maybe also should work on a complete route? 

Would be kind of weird. So maybe this *is* wrong. Maybe the testing facilities should "just" use the route/dispatching behavior that they're concerned with. But then the testing facilities do diverge slightly from the real world router/behavior...